### DISCUSSION TITLE
Getting SyntaxError: Named export 'createContext' not found on importing swiper inside dynamic pages in Next 14 production only

### DISCUSSION THREAD

**Discussion by Priyankcoder** [vercel/next.js#59992] [Help] [answered]
### Summary

I am using the next 14 page router and Swiper on the dynamic SSR page. It's working fine and as expected on local but it's throwing 500 internal server error on the Vercel production server. 
Error:
[
<img width="824" alt="Screenshot 2023-12-28 at 2 32 06 PM" src="https://github.com/vercel/next.js/assets/39910453/49b44616-4c27-46a4-8c75-6f2ebd1f2555">
](url)

Things I did to fix it but didn't work:
* Added "use client" in the component where I am importing swiper and in the dynamic page ([productName].js). Still getting the same error
* Tried every step mentioned here https://github.com/vercel/next.js/discussions/43755

Other folks are facing the same bug in multiple libraries
* https://github.com/vercel/next.js/issues/57582

I have been stuck on this and unable to resolve it for more than a week now 😅 . Let me know if any other information is required.

Any help would be really appreciated :smile

**Comment by ananuness** [comment]
There are some having this issue with next 14, as you mentioned, others are facing it too and no resolution were provided. What you can do is:

- Check your node version and update it if it's lower than 14;
- Update your swiper; or
- Downgrade your next.

**Reply by Priyankcoder** [reply, marked as answer]
Thanks @ananuness. I downgraded the next version to 13 and the node version to 18. It's working as expected now. 
[Here](https://blog.saeloun.com/2022/02/03/create-typescript-next-js-tailwind-application/) is a blog that I followed to downgrade to the next 13 just in case someone else need it

**Comment by Mockerman** [comment]
> npm i next@13.5
> npm i eslint-config-next@13

Helped me.
